src/services/api/routes/communities.ts

```diff
@@ -284,4 +302,122 @@ router.post('/', multer().none(), async function (request: express.Request, resp
 	}).end({ pretty: true, allowEmpty: true }));
 });
 
+router.post('/:community_action', multer().none(), async function (request: express.Request, response: express.Response): Promise<void> {
+	response.type('application/xml');
+
+	const parentCommunity: HydratedCommunityDocument | null = await getCommunityByTitleID(request.paramPack.title_id);
+	if (!parentCommunity) {
+		await respondCommunityNotFound(response);
+		return;
+	}
+
+	const communityActions: string[] = request.params.community_action.split('.');
+	const community: HydratedCommunityDocument | null = await commonGetSubCommunity(request, response, communityActions[0]);
+	if (!community) {
+		return;
+	}
+
+	if (communityActions.length > 1) {
+		if (communityActions[1] === 'delete') {
+			await deleteSubCommunity(community, request, response);
+		} else if (communityActions[1] === 'favorite') {
+			await addFavoriteSubCommunity(community, request, response);
+		} else if (communityActions[1] === 'unfavorite') {
+			await delFavoriteSubCommunity(community, request, response);
+		} else { // '%s.search'
+			response.sendStatus(501); // Not Implemented
+		}
+	} else {
+		await updateSubCommunity(community, request, response);
+	}
+});
+
+async function updateSubCommunity(community: HydratedCommunityDocument, request: express.Request, response: express.Response): Promise<void> {
+	response.type('application/xml');
+
+	if (community.owner != request.pid) {
+		response.sendStatus(403); // Forbidden
```

Comment:
Only commenting this once since it appears in other places as well. Is there an expected response body for this or just the status code? I'm unsure what the client actually expects in this case

Reply:
Sounds good, I appreciate the info!